packages/web3-core-method/src/index.js

```diff
@@ -755,13 +766,19 @@ Method.prototype.buildCall = function () {
                 if (gasPrice) {
                     payload.params[0].gasPrice = gasPrice;
                 }
+
                 sendRequest(payload, method);
             });
 
         } else {
             sendRequest(payload, method);
         }
 
+        if (isSendTx) {
+            setTimeout(() => {
+                defer.eventEmitter.emit('sent');
+            }, 10);
```

Comment:
Out of curiosity, does the test pass with a `0` length timeout? Would schedule the emit in the next tick...is that enough to resolve the race condition here?

Reply:
ah good idea I will give it a try with 0, and also see what happens with process.nextTick()

Reply:
set to `0` in https://github.com/ethereum/web3.js/pull/3550/commits/34bdd39aad540ccf03bd22f812ed2ec8584291b0 👍